server/routes/api/section.js

```diff
@@ -159,4 +165,45 @@ router.get('/updateSection', (req, res) => {
     });
 });
 
-module.exports = router;
\ No newline at end of file
+module.exports = router;
+
+// ------------- Combined Seating Layout and Course Methods -------------
+
+// gets information regarding a course given a student/prof id
+router.get('/getCourseView', (req, res) => {
+    // this is the professor ID
+    let courseID = req.body.courseID;
+
+    Course.findOne( {_id: courseID} , function(err, course){
+        if(err){
+            console.log(err);
+            return res.status(500).send();
+        }
+
+        seatingLayout = SeatingLayout.findOne({_id: course.seating_layout}, function(err, seatingLayout){
+
+            let myLayout;
+            if(err){
+                console.log(err);
+                return res.status(500).send();
+            }
+    
+            if(seatingLayout != null && seatingLayout.layout != null) {
+                myLayout = seatingLayout.layout;
+                console.log("?");
+            } else {
+                myLayout = "Layout not found";
+            }
+
+            return res.status(200).json({
+                "name" : course.name,
```

Comment:
The frontend would like more information like the **professor's name** and more importantly the **seating_arrangement** which is an array of students (with each student's id and name).

Reply:
yeah I agree. You could just return the whole `course` object instead of picking and choosing information.

Reply:
Note that even if we do return the whole course we still need to give the seating layout and professor's name separately since the course object will only contain their IDs rather than the actual professor/layout object.

Reply:
Ok, it should be returning all the required information now!